core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -502,6 +502,9 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     _applicationId = _taskScheduler.applicationId()
     _applicationAttemptId = taskScheduler.applicationAttemptId()
     _conf.set("spark.app.id", _applicationId)
+    if (_conf.getBoolean("spark.ui.reverseProxy", false)) {
+      System.setProperty("spark.ui.proxyBase", "/target/" + _applicationId)
```

Comment:
I am not sure if it would have any benefit, as currently it will look like /target/worker-Id or /target/app-id so url does have information about whether it is application or worker. Regarding proxy or target name, I have no preference. As to me it does convey that we are accessing worker/app page. I can make it a configurable value if that's preferred.